### PR TITLE
Replace almost all echos by printf function with color

### LINKED ISSUE #621: Use print functions instead of echo
Let's be consistent and use 3 `printf` `functions()` instead of `echo`.

```
printf_green() {
printf "%b" "${IGreen}$1\n${Color_Off}"
}

printf_red() {
printf "%b" "${IRed}$1\n${Color_Off}"
}

printf_normal() {
printf "%b" "${Color_Off}$1\n${Color_Off}"
}
```

# Example 
`printf_green hello`

Would print "hello" in green

### PR DESCRIPTION
**Pull request by ggeorgg**
I want to use a colored output for all echos from our scripts to make it easier to spot if an output is coming from our scripts or from other commands.
This should close #621 
Only the script nextcloud.sh and instructions.sh have echos left. I am not quite sure if we should replace them like all the other echos.

### CODE REVIEW

**Comment by enoch85** [comment]
WOW, Thanks!

This must be tested though. All the changes in all places.

File: nextcloud_install_production.sh [hunk 1]
```diff
@@ -35,7 +35,7 @@ fi
 # Install net-tools if not existing
 if [ "$(dpkg-query -W -f='${Status}' "net-tools" 2>/dev/null | grep -c "ok installed")" == "1" ]
 then
-    echo "net-tools OK"
+    print_text_in_color "$Cyan" "net-tools OK"
```
Comment (enoch85) [inline nextcloud_install_production.sh]: I think that when stuff are "OK" the text should be Green instead.

File: nextcloud_install_production.sh [hunk 2]
```diff
@@ -212,14 +212,14 @@ check_command apt install -y \
...
-echo "Enabling HTTP/2 server wide..."
+print_text_in_color "$Cyan" "Enabling HTTP/2 server wide..."
 cat << HTTP2_ENABLE > "$HTTP2_CONF"
 <IfModule http2_module>
     Protocols h2 h2c http/1.1
     H2Direct on
 </IfModule>
 HTTP2_ENABLE
-echo "$HTTP2_CONF was successfully created"
+print_text_in_color "$Cyan" "$HTTP2_CONF was successfully created"
```
Comment (enoch85) [inline nextcloud_install_production.sh]: Or when stuff are sucessfull, also Green.

File: static/modsecurity.sh [hunk 3]
```diff
@@ -93,6 +93,6 @@ check_command sed -i 's|# SecDefaultAction "phase:2,nolog,auditlog,pass"|SecDefa
 
 if [ -f /etc/modsecurity/whitelist.conf ]
 then
-    echo "ModSecurity activated!"
+    print_text_in_color "$Cyan" "ModSecurity activated!"
```
Comment (enoch85) [inline static/modsecurity.sh]: This is a sucessfull status, so Green.

**Comment by enoch85** [comment]
My five cents:

Informative: $Cyan
Successful or OK: $Green
Failed: $Red

There are more echos than in this PR, would be nice to change it everywhere. nextcloud.sh and instructions can be tested by running the certain script and put this function at the top.

**Comment by enoch85** [comment]
Tested some basic stuff, and it all seems to work. Good job @ggeorgg!

**Comment by enoch85** [comment]
Call for testing and help @nextcloud/vm

File: static/modsecurity.sh (hunk 3 above)
Comment (ggeorgg) [reply static/modsecurity.sh]: Totally agree. Thats also what I had in mind. Just wanted to make one change after another.

**Comment by ggeorgg** [comment]
Interessting for me would now be how to test these changes when they are not in master yet. The problem is that the lib is downloaded again for every single script and will then get the script from master branch.

Maybe we should set a default value for the variable $GITHUB_REPO but make it possible to change it with a name-value pair when executing the main script?

**Comment by enoch85** [comment]
> Maybe we should set a default value for the variable $GITHUB_REPO but make it possible to change it with a name-value pair when executing the main script?

Naah, then I think it's better if we use your repo as DEV and change GITHUB_REPO to that instead (including the curl command).

Usually I test it locally.

**Comment by enoch85** [comment]
So, the new repo would be `https://raw.githubusercontent.com/ggeorgg/vm/replace_echo`

**Comment by ggeorgg** [comment]
Yes I understand, but I don't like this idee because I would have to change the repo name in each file to test it. Let me think about another solution to be able to easily get another branch from this repository for test cases.

**Comment by enoch85** [comment]
I just tested this locally, and it seems to work. 

Will do one more test the merge.

**Comment by ggeorgg** [comment]
Sorry, I don't understand how test it locally. How do you avoid getting the scripts from master branch?

**Comment by enoch85** [comment]
>  scripts from master branch?

I change the curl command to the DEV repo.

**Comment by ggeorgg** [comment]
But you have to do it for all files, right?

**Comment by enoch85** [comment]
> But you have to do it for all files, right? 

Yes, usually I test one by one.

**Comment by ggeorgg** [comment]
ok, I understand. Thats something I didn't want to do.

**Comment by enoch85** [comment]
> ok, I understand. Thats something I didn't want to do.

Let me just finish up here, I'll merge this soon.

**Comment by enoch85** [comment]
OK, I'm satisfied. Will merge once the checks passed.

Thank you very much @ggeorgg! :D

**Comment by enoch85** [comment]
> Thats something I didn't want to do.

And yeah, regarding that. If you have a better idea I'm all ears!

**Comment by ggeorgg** [comment]
You are welcome ;)

**Pull request merged**